### PR TITLE
Adds script to run cmake unit tests.

### PR DESCRIPTION
**Pull request by Hsilgos**
Examples of usage:
cmake -P run-cmake-unit-test.cmake
GLUECODIUM_BUILD_ENVIRONMENT=ios-x86_64 CTEST_FILTER="test_to_run" cmake -P run-cmake-unit-test.cmake

See: #556 
Signed-off-by: Yauheni Khnykin <yauheni.khnykin@here.com>

### CODE REVIEW

File: cmake/run-cmake-unit-test.cmake [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env cmake -P
+# Copyright (C) 2016-2019 HERE Europe B.V.
```
Comment (DanielKamkha) [inline cmake/run-cmake-unit-test.cmake:2]: 2016-2020

File: cmake/tests/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Copyright (C) 2016-2019 HERE Europe B.V.
```
Comment (DanielKamkha) [inline cmake/tests/CMakeLists.txt:1]: 2016-2020

File: cmake/run-cmake-unit-test.cmake [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+# > GLUECODIUM_BUILD_ENVIRONMENT=ios-x86_64 CTEST_FILTER="example" ./run-cmake-unit-test.cmake
+
+set (TEST_BUILD_DIR "${CMAKE_CURRENT_LIST_DIR}/_test_build")
+set (OUTPUT_FILE cmake_unit_test_output.txt)
+set (DIRECTORY_WRITH_TESTS "${CMAKE_CURRENT_LIST_DIR}/tests")
+
+list (APPEND cmake_module_paths ${CMAKE_MODULE_PATH} "${DIRECTORY_WRITH_TESTS}")
+
+function (heresdk_configure_cmake)
+    file (MAKE_DIRECTORY ${TEST_BUILD_DIR})
```
Comment (DanielKamkha) [inline cmake/run-cmake-unit-test.cmake:39]: Is this dir deleted afterwards? Not important for CI, but important for running locally.

File: cmake/tests/run_tests.cmake [hunk 4]
```diff
@@ -0,0 +1,240 @@
+# Copyright (C) 2016-2019 HERE Europe B.V.
```
Comment (DanielKamkha) [inline cmake/tests/run_tests.cmake:1]: 2016-2020

File: cmake/tests/unit/example/example-XFAIL/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Copyright (C) 2016-2019 HERE Europe B.V.
```
Comment (DanielKamkha) [inline cmake/tests/unit/example/example-XFAIL/CMakeLists.txt:1]: 2016-2020

File: cmake/tests/unit/example/example_success/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Copyright (C) 2016-2019 HERE Europe B.V.
```
Comment (DanielKamkha) [inline cmake/tests/unit/example/example_success/CMakeLists.txt:1]: 2016-2020

File: cmake/tests/utils/message_colored.cmake [hunk 7]
```diff
@@ -0,0 +1,52 @@
+# Copyright (C) 2016-2019 HERE Europe B.V.
```
Comment (DanielKamkha) [inline cmake/tests/utils/message_colored.cmake:1]: 2016-2020

File: cmake/tests/utils/assert.cmake [hunk 8]
```diff
@@ -0,0 +1,604 @@
+# Copyright (C) 2016-2019 HERE Europe B.V.
```
Comment (DanielKamkha) [inline cmake/tests/utils/assert.cmake:1]: 2016-2020

**Review by DanielKamkha — CHANGES_REQUESTED** [review]

File: cmake/run-cmake-unit-test.cmake (hunk 3 above)
Comment (Hsilgos) [reply cmake/run-cmake-unit-test.cmake:39]: It's deleted before run and ignored by git. I guess leaving it undeleted might be helpfull when you want to investigate something after run

**Review by DanielKamkha — APPROVED** [review]

**Pull request merged**